python/lsst/obs/base/cameraMapper.py

```diff
@@ -420,16 +420,27 @@ def setMethods(suffix, mapImpl=None, bypassImpl=None, queryImpl=None):
                             setattr(self, addName, self.getImageCompressionSettings)
 
                         if name == "exposures":
-                            setMethods("wcs", bypassImpl=lambda datasetType, pythonType, location, dataId:
-                                       afwGeom.makeSkyWcs(readMetadata(location.getLocationsWithRoot()[0])))
+                            def getSkyWcs(datasetType, pythonType, location, dataId):
+                                fitsReader = afwImage.ExposureFitsReader(location.getLocationsWithRoot()[0])
+                                return fitsReader.readWcs()
+
+                            setMethods("wcs", bypassImpl=getSkyWcs)
+
                             setMethods("calib", bypassImpl=lambda datasetType, pythonType, location, dataId:
                                        afwImage.Calib(readMetadata(location.getLocationsWithRoot()[0])))
-                            setMethods("visitInfo",
-                                       bypassImpl=lambda datasetType, pythonType, location, dataId:
-                                       afwImage.VisitInfo(readMetadata(location.getLocationsWithRoot()[0])))
-                            setMethods("filter",
-                                       bypassImpl=lambda datasetType, pythonType, location, dataId:
-                                       afwImage.Filter(readMetadata(location.getLocationsWithRoot()[0])))
+
+                            def getVisitInfo(datasetType, pythonType, location, dataId):
+                                fitsReader = afwImage.ExposureFitsReader(location.getLocationsWithRoot()[0])
+                                return fitsReader.readVisitInfo()
+
+                            setMethods("visitInfo", getVisitInfo)
```

Comment:
Shouldn't this be `bypassImply=getVisitInfo`? Similarly for filter below.

Reply:
Good catch, thanks.